### ISSUE TITLE
feat(web): ask for consent before retaining an uploaded photograph for training

### ISSUE THREAD

**Issue by chuanseng-ng** [chuanseng-ng/tcg-analyzer#148]
Part of #7 (M6 — Dataset platform epic). Required by ADR 0008 (#67), risk **R5**.

## Purpose

ADR 0008 approves **this product's own user uploads** as a training-image source
*in principle*, and it supplies **zero images today** because nothing asks the
user for permission. Give it a consent mechanism, and give §54's retention sweep
the documented exception that retaining a consented image requires.

This is the only approved source that sits in the inference distribution at
volume. It scored 40/60 in the register with **two zeros of different kinds**:
`W1` ground-truth grade label is 0 permanently — the user is asking what grade
the card might get, so there is no label to have — and `W5` volume is 0 only
because this issue does not exist yet. That asymmetry is why the class is
approved in principle rather than rejected, and why it can never be the only
source.

## Scope

- A consent request in `apps/web`'s upload flow (`/analyze`), asked **before**
  anything is retained for training and separable from proceeding with the
  analysis. Declining must cost the user nothing.
- Consent recorded against the analysis, versioned by the text the user actually
  agreed to — §29's `license` for this class is *the consent text, by version*.
- A **documented exception** to §54's hourly retention sweep so a consented image
  survives its session's expiry, with the reasoning written into
  `docs/retention.md` beside the paragraph that already anticipates this.
- Withdrawal: how a user revokes, and what revocation can and cannot undo.

## Non-goals

- **No accounts and no login.** Spec §53 keeps sessions anonymous, and V1's
  exclusion list names authentication and user accounts outright. Whatever this
  is, it is not a sign-up.
- No ingestion into a dataset version — that is the ingestion issue's, once the
  provenance table exists.
- No change to what an unconsented image does. It is deleted on schedule, as now.

## Dependencies

ADR 0008 (#67), which approves the class and gates it on exactly this. The
provenance schema issue, if it lands first, owns where the consent is stored.

## Spec extract — §29, §53, §54 (verbatim)

> Every image needs:
>
> ```text
> source
> source_url/reference
> acquisition_method
> license
> commercial_use_allowed
> derivative_use_allowed
> redistribution_allowed
> permission_notes
> acquired_at
> ```
>
> Training pipeline should reject images where commercial-use status is unknown.

> Anonymous sessions. Do not permanently tie analyses to personal identity.

## Implementation notes

- **`docs/retention.md` already anticipated this and deferred it here**, in as
  many words: *"Retaining an image for training is a different question with a
  different answer. It is a separate, explicitly justified purpose governed by
  M6's provenance rules (§29): documented source, licence and commercial-use
  rights, per image. It must never happen because a retention sweep skipped
  something."* The last sentence is the design constraint — a consented image is
  retained because a **row says so**, never because the sweep failed to reach it.
- **Unknown is `false`.** ADR 0008's gate admits an image only where consent is
  positively recorded. A null, an empty string and an absent field are one
  answer, and it is refusal. Never default `commercial_use_allowed` to `true`.
- **Consent must name derivative use explicitly**, not just "use for improving
  the product". ADR 0008's interpretive rule 1 — silence is not a grant — binds a
  document this project wrote as firmly as anyone else's, and a trained model is
  a derivative work of its training data.
- **The sweep deletes the session row itself**, deliberately, so that a
  per-browser identifier is not kept forever. A consented image therefore cannot
  be reached through its session afterwards, and the exception has to carry
  whatever it needs at the moment of consent rather than by later lookup.
- Consider whether consent is asked per analysis or once per session. Per
  analysis is more honest about what is being agreed to; per session is fewer
  taps. Either way §53 forbids anything that survives as an identity.
- The user photographs their own hands, desks and rooms. That is exactly why the
  hardened `python-background-jobs` note refuses to keep payloads in a
  dead-letter queue, and the same reasoning applies to what a consent request
  should say plainly.

## Tests

- Consent absent, declined, and withdrawn each leave the image deletable on the
  normal schedule; only positively-recorded consent survives the sweep.
- The sweep's existing tests still pass unchanged for unconsented images.
- A consented image's nine §29 fields are all populated at the moment of upload,
  from the grantor, with `redistribution_allowed` **`false`**.

## Acceptance criteria

- A user can consent, decline, or withdraw, and declining costs them nothing.
- A consented image survives its session's expiry **because a row says so**, and
  an unconsented one does not.
- `docs/retention.md` documents the exception and why it exists.
- No account, no login, and nothing that ties an analysis to a person.

## Definition of Done — spec §71

- [ ] implementation complete
- [ ] tests added
- [ ] documentation updated
- [ ] error handling implemented
- [ ] logging appropriate
- [ ] API contracts documented
- [ ] no secrets committed
- [ ] reproducible locally
- [ ] acceptance criteria verified

## Conventions

- **One primary capability per PR.** The issue title is the intended commit subject.
- PR description uses these headings: Purpose, Scope, Non-goals, Dependencies, Implementation, Tests, Acceptance Criteria.
- Commit messages follow Conventional Commits; subject in imperative mood, lowercase, ≤72 characters, no trailing period.
- Never commit model weights, training images, API keys or provider credentials.

**Comment by chuanseng-ng** [comment]
Re-milestoned from **M6 — Dataset platform** to **M10 — V1 beta**.

The reason is not that this matters less. It is that it **supplies zero training images until the product has live users**, and its real gate is therefore the beta rather than M6's close:

- Approved class 4 is already a provenance value the ingestion CLI can be handed — `("product_upload", "uploaded_by_user_with_consent")` in `APPROVED_SOURCES` (`services/api/src/tcg_api/datasets/ingestion.py:130`) — so nothing about the *corpus* waits on this issue. What waits on it is a lawful upload from a real person, and there are none.
- The hardest part is still undesigned, and it is a beta-shaped question: **withdrawal with no accounts**. The sweep deliberately `DELETE`s the session row (`services/api/src/tcg_api/analysis/retention.py:171`, rationale in `docs/retention.md:53-61`) precisely so a per-browser identifier is not kept forever — which means a consented image cannot be reached through its session afterwards, and whatever the user is given to revoke with has to be handed over at the moment of consent. That is a §53 design decision, not a form field.
- ADR 0008's own framing agrees: class 4 is *"approved in principle and supplying nothing until a consent mechanism and a retention exception exist"*, and it scores 0 on volume **only because this issue does not exist yet**. That asymmetry does not decay while the product is offline.

The obligation is unchanged and the scope is unchanged. What changes is that it is now a **gate on shipping the beta** — no public upload flow ships without it — rather than an item holding M6 open indefinitely.

Refs: #7, #11, ADR 0008 (#67), spec §53, §54

**Comment by chuanseng-ng** [comment]
M10 was decomposed on 2026-09-05 (the comment on #11), and this issue now
**follows #270**.

The hardest part named above — *withdrawal with no accounts* — is the same
problem §68's feedback has: the session row is deleted on purpose, so
whatever a user is given to come back with has to be handed over at the
moment of consent. #270 builds that mechanism for the grade a user later
reports: a return code minted once, shown once, stored as a hash on a row
that carries no image, no session and no address, and expires on its own
clock, with a §54 exemption written into `docs/retention.md` before the
table.

This issue reuses it rather than designing a second one: a consent row mints
a return code with the same helper, the withdrawal page follows #274's
`/feedback/[code]` pattern, and this issue's retention exemption sits beside
#270's. What stays this issue's own: the consent question itself (naming
derivative use explicitly, ADR 0008's rule 1), the retention exception for the
**image**, the nine §29 fields at the moment of upload with
`redistribution_allowed` `false`, and what withdrawal can and cannot undo.

Order: #270 → #274 → this.
